src/olympia/addons/views.py

```diff
@@ -735,6 +736,39 @@ def get_queryset(self):
         return qs.filter(addon=self.get_addon_object())
 
 
+class AddonReviewNotesViewSet(RetrieveModelMixin, ListModelMixin,
+                              GenericViewSet):
+    class AddonViewSetNoPublic(AddonViewSet):
+        permission_classes = [
+            AnyOf(AllowAddonAuthor, AllowReviewer, AllowReviewerUnlisted),
+        ]
+    permission_classes = [
+        AnyOf(AllowAddonAuthor, AllowReviewer, AllowReviewerUnlisted),
+    ]
+    serializer_class = ReviewNotesSerializer
+    queryset = Version.unfiltered.all()
+
+    def get_addon_object(self):
```

Comment:
it's subtly different - it's using `AddonViewSetNoPublic` instead of `AddonViewSet` to get the permissions correct.  I can still generalise if worth it?